File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        int evens = 0;
        int odds = 0;
        Arrays.sort(nums1);
        int min = Integer.MAX_VALUE;
        for(int val : nums1){
            min = Math.min(min,val);
            if(val%2==0){
                evens++;
            }else{
                odds++;
            }
        } 
        if(evens == nums1.length){
            return true;
        }
        if(odds == nums1.length){
            return true;
        }
        if(min%2==0){
           return false;
        }else{
           return true;
        }

    }
}

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.py
class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        min_val = float('inf')
        hasodd = False
        haseven = False

        for val in nums1:
            min_val = min(val,min_val)
            if val % 2 == 0:
                haseven = True
            else:
                hasodd = True

        if not hasodd or not haseven:
            return True

        return min_val % 2!=0  